saleor/order/calculations.py

```diff
@@ -0,0 +1,261 @@
+from decimal import Decimal
+from typing import Iterable, Optional, Tuple
+
+from django.conf import settings
+from django.db.models import prefetch_related_objects
+from django.utils import timezone
+from prices import Money, TaxedMoney
+
+from ..core.prices import quantize_price
+from ..core.taxes import TaxData, TaxError, TaxLineData, zero_taxed_money
+from ..plugins.manager import PluginsManager
+from . import ORDER_EDITABLE_STATUS
+from .models import Order, OrderLine
+
+
+def _combine_prices_from_manager_into_tax_data(
+    manager: PluginsManager, order: Order, lines: Iterable[OrderLine]
+) -> Optional[TaxData]:
+    prefetch_related_objects(lines, "variant__product")
```

Comment:
Is it expected that this will work only partially? For some methods that calls `update_order_prices_if_expired` you don't pass a lines, which means that it is None, and then you're calling `lines = list(order.lines.all())`, which means that we will not use any benefit of prefetching a data.

Reply:
Resolved in a recent commit.